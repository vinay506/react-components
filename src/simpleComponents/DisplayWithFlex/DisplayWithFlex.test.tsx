import React from 'react';
import { shallow } from 'enzyme';
import { DisplayWithFlex, RoundedBox, NameWithSubText } from '../index';
import renderer from 'react-test-renderer'
import 'jest-styled-components'


describe('<Display With Flex >', function () {
    let wrapper: any;
    let props: any;
    let tree: any;
    beforeEach(() => {
        props = {

        };
        tree = renderer.create(<DisplayWithFlex {...props}></DisplayWithFlex>).toJSON()
    });

    // Snapshot test
    it('should render', () => {
        expect(tree).toMatchSnapshot()
    })


    it('checks the broder radius ', () => {
        expect(tree).toHaveStyleRule('display', 'flex')
    });

    it('checks the Childs Count ', () => {
        wrapper = shallow(
            <DisplayWithFlex {...props}>
                <RoundedBox style={{
                    height: '30px',
                    width: '30px',
                    marginBottom: '20px',
                    marginRight: '10px'
                }}>
                </RoundedBox>
                <NameWithSubText nameInfo={{
                    name: 'Asha Sharma',
                    subText: '60 AIR'
                }}>
                </NameWithSubText>

            </DisplayWithFlex>
        );

        const childrenCount = wrapper.children();
        expect(childrenCount).toHaveLength(2)

    })

})