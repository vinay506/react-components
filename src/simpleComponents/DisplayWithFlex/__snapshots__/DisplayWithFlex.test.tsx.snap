// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display With Flex > should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;
